### PR TITLE
Fix some case of #6683

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

### What problem does this PR solve?

This PR is to fix corner case of https://github.com/tikv/tikv/pull/6683
And add some unit test.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 
- No release note

### CODE REVIEW

File: components/raftstore/src/store/fsm/peer.rs [hunk 1]
```diff
@@ -270,7 +270,10 @@ impl BatchRaftCmdRequestBuilder {
         // No batch request whose size exceed 20% of raft_entry_max_size,
         // so total size of request in batch_raft_request would not exceed
         // (40% + 20%) of raft_entry_max_size
-        if req.has_admin_request() || f64::from(req_size) > self.raft_entry_max_size * 0.2 {
+        if req.has_admin_request()
+            || req.has_status_request()
```
Comment (gengliqi) [inline components/raftstore/src/store/fsm/peer.rs:274]: What if another new type is added to `RaftCmdRequest`? How about `!req.has_requests()`?

Comment (Little-Wallace) [reply components/raftstore/src/store/fsm/peer.rs:274]: Good idea.

File: components/raftstore/src/store/fsm/peer.rs [hunk 2]
```diff
@@ -270,7 +270,7 @@ impl BatchRaftCmdRequestBuilder {
         // No batch request whose size exceed 20% of raft_entry_max_size,
         // so total size of request in batch_raft_request would not exceed
         // (40% + 20%) of raft_entry_max_size
-        if req.has_admin_request() || f64::from(req_size) > self.raft_entry_max_size * 0.2 {
+        if 0 == req.get_requests().len() || f64::from(req_size) > self.raft_entry_max_size * 0.2 {
```
Comment (BusyJay) [inline components/raftstore/src/store/fsm/peer.rs:273]: Better `!req.get_requests().is_empty()`.

File: components/raftstore/src/store/fsm/peer.rs [hunk 3]
```diff
@@ -3411,3 +3411,99 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
...
+    use std::sync::atomic::{AtomicBool, Ordering};
+    use std::sync::Arc;
+
+    #[test]
+    fn test_batch_raft_cmd_request_builder() {
+        let max_batch_size = 1000.0;
+        let mut builder = BatchRaftCmdRequestBuilder::new(max_batch_size);
+        let mut q = Request::default();
+        let mut metric = RaftProposeMetrics::default();
+        {
```
Comment (BusyJay) [inline components/raftstore/src/store/fsm/peer.rs:3435]: We don't use brackets in rust tests. Just blank line is OK.

Comment (Little-Wallace) [reply components/raftstore/src/store/fsm/peer.rs]: Ok.

**Pull request merged**